Sources/performance-worker/WorkerService.swift

```diff
@@ -98,3 +134,53 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
     throw RPCError(code: .unimplemented, message: "This RPC has not been implemented yet.")
   }
 }
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension WorkerService {
+  private func setupServer(_ config: Grpc_Testing_ServerConfig) async throws {
+    try self.state.withLockedValue { state in
+      switch state.role {
+      case .server(_):
+        throw RPCError(code: .alreadyExists, message: "A server has been previously set up.")
+      case .client(_):
+        throw RPCError(code: .failedPrecondition, message: "This worker has a client set up.")
+      default:
+        let server = GRPCServer(transports: [], services: [BenchmarkService()])
+        state.role = .server(server)
+      }
+    }
+    let initialStats = try await ServerStats()
+    self.state.withLockedValue {
+      $0.initialStats = initialStats
+    }
+  }
+
+  private func makeStatsResponse(
+    for mark: Grpc_Testing_Mark
```

Comment:
`for mark: Grpc_Testing_Mark` isn't particularly helpful here, `reset: Bool` is clearer.